### PR TITLE
fix(lime-plugin-loader.interface): make deprecated `componentDidUnload` method optional

### PR DESCRIPTION
**Pull request by adrianschmidt**

### CODE REVIEW

File: packages/interfaces/src/lime-plugin-loader.interface.ts [hunk 1]
```diff
@@ -27,13 +27,12 @@ export interface LimePluginLoader extends LimeWebComponent {
...
      * This method will never be called since the component will live
-     * during the whole lifespan of the application. It must still be
-     * present on the component though.
+     * during the whole lifespan of the application.
      *
      * @deprecated this method has been deprecated and consumers should
      * use `disconnectedCallback` instead
      */
-    componentDidUnload(): void | Promise<any>;
+    componentDidUnload?(): void | Promise<any>;
```
Comment (adrianschmidt) [inline packages/interfaces/src/lime-plugin-loader.interface.ts:35]: If this is still required for technical reasons, we need to update that technical reason, as this method has been removed in Stencil 2.

Comment (jgroth) [reply packages/interfaces/src/lime-plugin-loader.interface.ts:35]: It should not be required, so this should be safe :slightly_smiling_face: 
Let's keep it around for some time before we remove it and make a breaking change

**Review by jgroth — APPROVED** [review]

File: packages/interfaces/src/lime-plugin-loader.interface.ts (hunk 1 above)
Comment (adrianschmidt) [reply packages/interfaces/src/lime-plugin-loader.interface.ts:35]: Yeah, I agree 👍

When we do, we should probably make `disconnectedCallback` mandatory as well?

Comment (jgroth) [reply packages/interfaces/src/lime-plugin-loader.interface.ts:35]: Yes, and `connectedCallback` :slightly_smiling_face: 
I think there are a few other breaking changes as well that I have saved up :smile:

**Pull request merged**